### PR TITLE
fix: crash when user doesn't pass icon props to  message.open

### PR DESCRIPTION
**Pull request by AshoneA**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/issues/21745

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    fix crash when the user doesn't pass `icon` props to `message.open`       |
| 🇨🇳 Chinese |      修复了`message.open`中`icon`为空的时候报错    |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/message/index.tsx [hunk 1]
```diff
@@ -93,7 +93,7 @@ function notice(args: ArgsProps): MessageType {
               args.type ? ` ${prefixCls}-${args.type}` : ''
             }`}
           >
-            {args.icon ? args.icon : <IconComponent />}
+            {args.icon || (IconComponent && <IconComponent />)}
```
Comment (yoyo837) [inline components/message/index.tsx]: 这改变了原逻辑吧？指定的怎么显示？

Comment (yoyo837) [reply components/message/index.tsx]: 不应该是type的问题吗？

Comment (AshoneA) [reply components/message/index.tsx]: `args.icon`不为空的时候还是显示`args.icon`， `IconComponent`有值的时候才会显示`<IconComponent />`。message.open中type不是用户传的呀。

Comment (AshoneA) [reply components/message/index.tsx]: type是内部方法的参数， 比如`message.success`时候才会传`type`

**Pull request merged**